storage/storage.go

```diff
@@ -1068,25 +1068,32 @@ func (o *ObjectHandle) SoftDeleted() *ObjectHandle {
 	return &o2
 }
 
-// Restore will restore a soft-deleted object to full object status.
+// RestoreOptions allows you to set options when restoring an object.
+type RestoreOptions struct {
+	/// CopySourceACL indicates whether the restored object should copy the
+	// access controls of the source object. Only valid for buckets with
+	// fine-grained access. If uniform bucket-level access is enabled, setting
+	// CopySourceACL will cause an error.
+	CopySourceACL bool
+}
+
+// Restore will restore a soft-deleted object to a live object.
 // Note that you must specify a generation to use this method.
```

Comment:
If generation is not specified, do we get an intelligible error from the server based on whatever the Go client sends by default? Might be good to check this.

Reply:
The error should be intelligible, let me check...

GRPC
`rpc error: code = InvalidArgument desc = You must specify a generation. error details: name = BadRequest field =  desc = You must specify a generation.`

Ah, HTTP's is no longer intelligible since it's a required param ` googleapi: Error 400: Invalid argument., invalid`

What do you suggest we do here? Set `if o.gen < 0 { gen = 0 }` to get the intelligible error? Sending generation=0 results in `googleapi: Error 400: Restore request must specify a generation., required`

Reply:
I think sending zero to get the intelligible error seems best here